go/libraries/doltcore/mvdata/data_loc.go

```diff
@@ -65,38 +83,38 @@ func (df DataFormat) ReadableStr() string {
 	}
 }
 
-func DFFromString(dfStr string) DataFormat {
-	switch strings.ToLower(dfStr) {
-	case "csv", ".csv":
-		return CsvFile
-	case "psv", ".psv":
-		return PsvFile
-	case "xlsx", ".xlsx":
-		return XlsxFile
-	case "json", ".json":
-		return JsonFile
-	case "sql", ".sql":
-		return SqlFile
-	default:
-		return InvalidDataFormat
-	}
-}
+// DataLocation is an interface that can be used to read or write from the source or the destination of a move operation.
+type DataLocation interface {
+	fmt.Stringer
 
-type DataLocation struct {
-	Path   string
-	Format DataFormat
-}
+	// Exists returns true if the DataLocation already exists
+	Exists(ctx context.Context, root *doltdb.RootValue, fs filesys.ReadableFS) (bool, error)
 
-func (dl *DataLocation) String() string {
-	return dl.Format.ReadableStr() + ":" + dl.Path
-}
+	// NewReader creates a TableReadCloser for the DataLocation
+	NewReader(ctx context.Context, root *doltdb.RootValue, fs filesys.ReadableFS, schPath string, opts interface{}) (rdCl table.TableReadCloser, sorted bool, err error)
 
-func NewDataLocation(path, fileFmtStr string) *DataLocation {
-	dataFmt := InvalidDataFormat
+	// NewCreatingWriter will create a TableWriteCloser for a DataLocation that will create a new table, or overwrite
+	// an existing table.
+	NewCreatingWriter(ctx context.Context, mvOpts *MoveOptions, root *doltdb.RootValue, fs filesys.WritableFS, sortedInput bool, outSch schema.Schema, statsCB noms.StatsCB) (table.TableWriteCloser, error)
 
-	if fileFmtStr == "" {
+	// NewUpdatingWriter will create a TableWriteCloser for a DataLocation that will update and append rows based on
+	// their primary key.
+	NewUpdatingWriter(ctx context.Context, mvOpts *MoveOptions, root *doltdb.RootValue, fs filesys.WritableFS, srcIsSorted bool, outSch schema.Schema, statsCB noms.StatsCB) (table.TableWriteCloser, error)
+}
+
+// NewDataLocation creates a DataLocation object from a path and a format string.  If the path is the name of a table
+// then a TableDataLocation will be returned.  If the path is empty a StdioDataLocationo is returned.  Otherwise a
```

Comment:
typo StdioDataLocation

Reply:
Also looks like you renamed this to StreamDataLocation since this comment

Reply:
fixed